### PR TITLE
Validate the shape of each proof

### PR DESCRIPTION
**Pull request by dlubarov**

### CODE REVIEW

File: plonky2/src/fri/validate_shape.rs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+            steps,
+        } = query_round;
+
+        ensure!(initial_trees_proof.evals_proofs.len() == instance.oracles.len());
+        for ((leaf, merkle_proof), oracle) in initial_trees_proof
+            .evals_proofs
+            .iter()
+            .zip(&instance.oracles)
+        {
+            ensure!(leaf.len() == oracle.num_polys); // TODO: Account for blinding if ZK?
```
Comment (wborgeaud) [inline plonky2/src/fri/validate_shape.rs:43]:
```suggestion
            ensure!(
                leaf.len()
                    == oracle.num_polys
                        + if oracle.blinding && params.hiding {
                            SALT_SIZE
                        } else {
                            0
                        }
            );
```

File: plonky2/src/plonk/validate_shape.rs [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    ensure!(wires_cap.height() == cap_height);
+    ensure!(plonk_zs_partial_products_cap.height() == cap_height);
+    ensure!(quotient_polys_cap.height() == cap_height);
+
+    ensure!(openings.constants.len() == common_data.num_constants);
+    ensure!(openings.plonk_sigmas.len() == config.num_routed_wires);
+    ensure!(openings.wires.len() == config.num_wires);
+    ensure!(openings.plonk_zs.len() == config.num_challenges);
+    ensure!(openings.plonk_zs_next.len() == config.num_challenges);
+    ensure!(openings.partial_products.len() == common_data.num_partial_products);
```
Comment (wborgeaud) [inline plonky2/src/plonk/validate_shape.rs:63]:
```suggestion
    ensure!(
        openings.partial_products.len() == config.num_challenges * common_data.num_partial_products
    );
```

**Review by wborgeaud — APPROVED** [review]
Looks great! I think we should go with this approach. It makes the verifier safer at little cost.

**Comment by dlubarov** [comment]
Thanks for the fixes!

**Pull request merged**